src/vs/base/browser/markdownRenderer.ts

```diff
@@ -115,6 +115,32 @@ const defaultMarkedRenderers = Object.freeze({
 	},
 });
 
+/**
+ * Process markdown alerts in rendered HTML
+ * Transforms blockquotes with alert syntax like "[!NOTE]" into structured alert markup
+ */
+function processAlertsHtml(html: string): string {
+	// Map alert types to their codicons
+	const alertIcons: Record<string, string> = {
+		'NOTE': 'info',
+		'TIP': 'light-bulb',
+		'IMPORTANT': 'comment',
+		'WARNING': 'alert',
+		'CAUTION': 'stop'
+	};
+
+	// Match blockquotes that start with [!TYPE]
+	const alertRegex = /<blockquote>\s*<p>\[!(NOTE|TIP|IMPORTANT|WARNING|CAUTION)\](\s*(?:<br\/?>|[\r\n]))/gi;
```

Comment:
I just learned about named capturing groups a year or so ago: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group

Reply:
Good suggestion -- though it turns out that `String.replace()` takes the groups as its last argument, so we would just have even more ignored parameters in the implementation below. So I think it's still cleaner this way.